user/src/main/java/com/scaler/user/services/UserServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.scaler.user.services;
+
+import com.scaler.user.dto.UserDTO;
+import com.scaler.user.exceptionsHandler.NotFoundException;
+import com.scaler.user.models.User;
+import com.scaler.user.repository.UserRepo;
+import com.scaler.user.utilities.RoleType;
+import jakarta.transaction.Transactional;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepo userRepo;
+
+    public UserServiceImpl(UserRepo userRepo) {
+        this.userRepo = userRepo;
+    }
+
+    @Override
+    public User registerUser(UserDTO userDTO) throws NotFoundException {
+        if (userRepo.existsByUsername(userDTO.getUsername()) || userRepo.existsByEmail(userDTO.getEmail())) {
+            throw new NotFoundException("Username is already in use");
+        }
+        User user = new User();
+        user.setFullName(userDTO.getFullName());
+        user.setUsername(userDTO.getUsername());
+        user.setEmail(userDTO.getEmail());
+        user.setPassword(userDTO.getPassword());
```

Comment:
I think you need to encrypt the password before saving it for security purposes.